### PR TITLE
Remove cromwell.binding and refactor to reflect new wdl4s binding rea…

### PR DESCRIPTION
**Pull request by geoffjentry**
…lity

Notes: 
- As w/ the WDL half of this, I will not be accepting passenger requests for random fixups but only on the actual changes (should be easier here)
- sbt assembly is borked, I know that
- currently loading in wdl4s as an unmanaged jar, that'll change prior to merging

### CODE REVIEW

**Comment by geoffjentry** [comment]
The Wheel of R hath chosen @kshakir @Horneth

File: README.md [hunk 1]
```diff
@@ -409,396 +333,9 @@ $ java -jar cromwell.jar highlight test.wdl html
...
-The inputs required to run this would be:
-
-```
-{
-  "test.hello.name": "world",
-  "test.hello2.name": "boston",
-  "test.greeting": "hello"
-}
-```
+# Getting Started with WDl
```
Comment (kshakir) [inline README.md]: `WDl` -> `WDL`

File: README.md [hunk 2]
```diff
@@ -409,396 +333,9 @@ $ java -jar cromwell.jar highlight test.wdl html
...
-This example calls the `analysis` task once for each element in the array that the `prepare` task outputs.  The resulting outputs of this workflow would be:
-
-```
-{
-  "example.analysis.out": ["_one_", "_two_", "_three_", "_four_"],
-  "example.gather.str": "_one_ _two_ _three_ _four_",
-  "example.prepare.array": ["one", "two", "three", "four"]
-}
-```
+For many examples on how to use WDL see PUT A LINK HERE
```
Comment (kshakir) [inline README.md]: Put a link there

Comment (geoffjentry) [inline README.md]: :laughing:

**Comment by kshakir** [comment]
Why is `lib/wdl4s-0.1.jar` committed into a source repo? Should be in an artifact repo.

**Comment by geoffjentry** [comment]
@kshakir From the PR note: "currently loading in wdl4s as an unmanaged jar, that'll change prior to merging"

File: src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala [hunk 3]
```diff
@@ -166,12 +167,18 @@ trait SharedFileSystem {
   /**
    * Return a possibly altered copy of inputs reflecting any localization of input file paths that might have
    * been performed for this `Backend` implementation.
-   */
-  def adjustInputPaths(callKey: CallKey, inputs: CallInputs, workflowDescriptor: WorkflowDescriptor): CallInputs = {
+    *
+    *     * NOTE: This ends up being a backdoor implementation of Backend.adjustInputPaths as both LocalBackend and SgeBackend
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala]: Never seen a note prefixed with ` * NOTE: `. Just checking if this was an intentional style preference and not a copy / paste issue?

File: src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java [hunk 4]
```diff
@@ -12,4 +12,4 @@
         this.diskTypeName = diskTypeName;
         this.googleTypeName = googleTypeName;
     }
-}
+}
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java]: Put the `\n` back

File: src/main/scala/cromwell/engine/backend/runtimeattributes/RuntimeKey.java [hunk 5]
```diff
@@ -64,4 +64,4 @@ public boolean isOptional(BackendType backendType) {
     public boolean supports(BackendType backendType) {
         return isOptional(backendType) || isMandatory(backendType);
     }
-}
+}
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/RuntimeKey.java]: ditto

File: src/main/scala/cromwell/engine/io/IoInterface.scala [hunk 6]
```diff
@@ -17,4 +17,4 @@ trait IoInterface {
   def hash(path: String): String
 
   def isValidPath(path: String): Boolean
-}
+}
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/io/IoInterface.scala]: Revert

File: src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java (hunk 4 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java]: Huh, this must have been intellij or something else like that

File: src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala (hunk 3 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala]: That was definitely some form of an editing fail

**Comment by kshakir** [comment]
Minor updates, then LGTM

File: src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java (hunk 4 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/DiskType.java]: Oh I know, these classes were truly copypasta

File: README.md [hunk 7]
```diff
@@ -411,394 +335,7 @@ Start a server on port 8000, the API for the server is described in the [REST AP
...
-This example calls the `analysis` task once for each element in the array that the `prepare` task outputs.  The resulting outputs of this workflow would be:
-
-```
-{
-  "example.analysis.out": ["_one_", "_two_", "_three_", "_four_"],
-  "example.gather.str": "_one_ _two_ _three_ _four_",
-  "example.prepare.array": ["one", "two", "three", "four"]
-}
-```
+For many examples on how to use WDL see [the WDL site](https://github.com/broadinstitute/wdl/tree/develop#getting-started-with-wdl)
```
Comment (geoffjentry) [inline README.md]: For reviewers note that this is intended as a placeholder link. The eventual home will be somewhere on the official site once that's rolled out in a couple of weeks

**Comment by kshakir** [comment]
Whoops! Thanks for the pointer

**Comment by geoffjentry** [comment]
@kshakir Good thing you saw it cuz now you won't be surprised when I ask you to hold my hand on monday morning ;)

**Comment by Horneth** [comment]
LGTM too

**Pull request merged**